### PR TITLE
action: add webhook test

### PR DESCRIPTION
**Pull request by Desiki-high**
We should test if webhook work happy.

### CODE REVIEW

**Comment by Desiki-high** [comment]
@imeoer It seems that the webhook task is triggered but the task work error occurs. Does the webhook work well in the main branch? Why `acceld` don't receive the webhook request?

image

**Comment by Desiki-high** [comment]
> @imeoer It seems that the webhook task is triggered but the task work error occurs. Does the webhook work well in the main branch? Why `acceld` don't receive the webhook request?
> 
> image

The webhook config is wrong!

File: .github/workflows/webhook-test.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    branches: [ '*' ]
+    paths-ignore: [ '**.md']
+  pull_request:
+    branches: [ main ]
+    paths-ignore: [ '**.md']
+
+env:
+  GO_VERSION: 1.18
+  HARBOR_VERSION: 2.8.4
+  NYDUS_VERSION: 2.1.6
```
Comment (imeoer) [inline .github/workflows/webhook-test.yml:14]: Maybe update to `2.2.3`

File: script/integration/webhook/test.sh [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+while [ $count -lt 60 ]; do
+  response=$(curl -b "" --location 'http://0.0.0.0/api/v2.0/projects/library/repositories/alpine/artifacts/latest-nydus/tags?page=1&page_size=10&with_signature=false&with_immutable_status=false' \
+  --header 'accept: application/json' \
+  --header 'X-Request-Id: 1' \
+  -s -o /dev/null -w "%{http_code}")
+
+  cat acceld.log
+  if [ "$response" == "200" ]; then
+    curl -b "" --location 'http://0.0.0.0/api/v2.0/projects/library/repositories/alpine/artifacts/latest-nydus/tags?page=1&page_size=10&with_signature=false&with_immutable_status=false' --header 'accept: application/json'
+    if grep -E 'level=error|panic:' acceld.log; then
```
Comment (imeoer) [inline script/integration/webhook/test.sh:66]: It seems too strict (panic) to check acceld service here.

File: script/integration/webhook/test.sh [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      \"event_types\": [
+        \"PUSH_ARTIFACT\"
+      ],
+      \"creator\": \"admin\",
+      \"creation_time\": \"2023-09-19T03:54:32.967Z\",
+      \"update_time\": \"2023-09-19T03:54:32.967Z\",
+      \"enabled\": true
+    }")
+
+    if [[ "$response" != *"<title>502 Bad Gateway</title>"* ]]; then
```
Comment (imeoer) [inline script/integration/webhook/test.sh:42]: Can we just check the status code? Should be `2xx`.

File: .github/workflows/webhook-test.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    branches: [ '*' ]
+    paths-ignore: [ '**.md']
+  pull_request:
+    branches: [ main ]
+    paths-ignore: [ '**.md']
+
+env:
+  GO_VERSION: 1.18
+  HARBOR_VERSION: 2.8.4
+  NYDUS_VERSION: 2.2.3
```
Comment (Desiki-high) [inline .github/workflows/webhook-test.yml:14]: It also failed with v2.3.0-alpha.0.

**Review by imeoer — APPROVED** [review]

**Pull request merged**